src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs

```diff
@@ -188,7 +188,7 @@ private protected override void SaveAsOnnxCore(OnnxContext ctx)
             Host.Assert(Bindings.InfoCount >= 2);
 
             if (!ctx.ContainsColumn(DefaultColumnNames.Features))
-                return;
+                throw Contracts.ExceptParam(nameof(DefaultColumnNames.Features), $"Please use default Column Names: '{DefaultColumnNames.Features}'");
 
```

Comment:
The user is free to choose the column name for their features. So we need to extract the right column name. (e.g. See on line 202 where we get the scoreColumn) You cannot do it exactly that way because the Features column index is not stored alongside the score column.  Please look up `RoleMappedSchema`  and `ISchemaBoundMapper`

Reply:
Yeah, I agree, seems this behavior is enabled for regression (support for non-default features column name), we better keep behavior consistent if possible.

---
In reply to: [428299686](https://github.com/dotnet/machinelearning/pull/5146#discussion_r428299686) [](ancestors = 428299686)

Reply:
> Yeah, I agree, seems this behavior is enabled for regression (support for non-default features column name), we better keep behavior consistent if possible.
> 
> In reply to: [428299686](https://github.com/dotnet/machinelearning/pull/5146#discussion_r428299686) [](ancestors = 428299686)

any link to what "regression" is referring to?

Reply:
> The user is free to choose the column name for their features. So we need to extract the right column name. (e.g. See on line 202 where we get the scoreColumn) You cannot do it exactly that way because the Features column index is not stored alongside the score column. Please look up `RoleMappedSchema` and `ISchemaBoundMapper`

Does that mean when user uses non-default name we need to 'translate' to its default name since those default names are used under the hood?

Reply:
No, I mean you should not throw an exception if the user has chosen a different name for their Features column. I am not sure why that check for `ctx.ContainsColumn(DefaultColumnNames.Features)` is present because the `SaveAsOnnxCore` function is not relying on it. Maybe the `base.SaveAsOnnxCore(ctx)` call is relying on it. 

The bug #3089 is not very clear on what the exact failure is. So please add a test in `OnnxConversionTest.cs` as @frank-dong-ms suggested to first see what the behavior is. 

---
In reply to: [428376984](https://github.com/dotnet/machinelearning/pull/5146#discussion_r428376984) [](ancestors = 428376984)

Reply:
I am guessing @frank-dong-ms means that this will break previous behavior and cause a regression if the user is not allowed to choose a different name for the Features column and export the model.

---
In reply to: [428376134](https://github.com/dotnet/machinelearning/pull/5146#discussion_r428376134) [](ancestors = 428376134)

Reply:
> No, I mean you should not throw an exception if the user has chosen a different name for their Features column. I am not sure why that check for `ctx.ContainsColumn(DefaultColumnNames.Features)` is present because the `SaveAsOnnxCore` function is not relying on it. Maybe the `base.SaveAsOnnxCore(ctx)` call is relying on it.
> 
> The bug #3089 is not very clear on what the exact failure is. So please add a test in `OnnxConversionTest.cs` as @frank-dong-ms suggested to first see what the behavior is.
> 
> In reply to: [428376984](https://github.com/dotnet/machinelearning/pull/5146#discussion_r428376984) [](ancestors = 428376984)

I just find that if those two lines(ctx.ContainsColumn(DefaultColumnNames.Features) return;) are removed there is no issue any more. Ok I will add a test for that